src/main/webapp/scripts/place-suggestions.js

```diff
@@ -1,38 +1,49 @@
-let map;
-let service;
-
-const MTV = new google.maps.LatLng(37.3861,122.0839);
-
-/**
- * Inits map and calls a nearby search of prominent restaurants in 5000m 
- * radius of MTV
- */
-function findNearbyPlaces() {
-  map = new google.maps.Map(document.getElementById('map'), {
-      center: MTV,
-      zoom: 15
-    });
+let city;
+let filter;
 
-  let request = {
-    location: MTV,
-    radius: '5000',
-    type: ['restaurant']
-  };
+function init() {
+  authReload();
+  addCityAutocomplete();
+  addFilterHandler();
+}
 
-  service = new google.maps.places.PlacesService(map);
-  service.nearbySearch(request, callback);
+function addCityAutocomplete() {
+  const cityInputField = document.getElementById('city-input')
+  const options = {
+    types: ['(cities)'],
+  }
+  const autocomplete = new google.maps.places.Autocomplete(cityInputField, options);
+  google.maps.event.addListener(autocomplete, 'place_changed', () => {
+    city = autocomplete.getPlace();
+    if(typeof filter !== 'undefined') {
+      findPlacesInCity(city, filter);
+    }
+  });
 }
 
-/**
- * Logs the name of each place returned by nearby search in findNearbyPlaces()
- * to browser console
- * @param {Object} results array of PlaceResults objects
- * @param {Object} status status of Places API call
- */
-function callback(results, status) {
-  if (status === google.maps.places.PlacesServiceStatus.OK) {
-    for (let i = 0; i < results.length; i++) {
-      console.log(results[i].name);
+function addFilterHandler() {
+  const filterForm = document.getElementById('filters');
+  filterForm.addEventListener('change', () => {
+    filter = filterForm.value;
+    if(typeof city !== 'undefined') {
+      findPlacesInCity(city, filter);
     }
+  });
+}
+
+function findPlacesInCity(city, filter) {
+  if (city.hasOwnProperty('place_id') === false) {
```

Comment:
Why not use "!city.hasOwnProperty('place_id')" ?

Reply:
Will fix, I'm out of practice with js conditionals